### PR TITLE
  **Breaking:** Change accordion API

### PR DESCRIPTION
**Pull request by stereobooster**
# Summary

Change accordion API as discussed with @fabien0102 

# Related issue

UI-77

# To be tested

Me
- [ ] No error or warning in the console on `localhost:6060`

Tester 1

- [ ] Things look good on the demo.
  

Tester 2

- [ ] Things look good on the demo.

### CODE REVIEW

File: src/AccordionSection/AccordionSection.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+import styled from "../utils/styled"
+import { ChevronUpIcon, ChevronDownIcon } from "../Icon/Icon"
+import { DefaultProps } from "../types"
+import isFunction from "lodash/isFunction"
+
+export interface AccordionSectionProps extends DefaultProps {
+  title: React.ReactNode
+  children: (() => React.ReactNode) | React.ReactNode
+  // for internal use
+  expanded: boolean
```
Comment (stereobooster) [inline src/AccordionSection/AccordionSection.tsx]: @fabien0102 shall I name those differently? I can them with the help of Context API to not expose to end user

**Comment by stereobooster** [comment]
@fabien0102 I saw it. Do you know what is this? I can't spot the issue

File: src/Accordion/README.md [hunk 2]
```diff
@@ -10,7 +10,7 @@ import { Accordion, AccordionSection } from "@operational/components"
 
 const MyComponent = () => {
   const [expanded, setExpanded] = React.useState([true, false, false])
-  const onToggle = (index: number) => {
+  const onToggle = index => {
```
Comment (TejasQ) [inline src/Accordion/README.md]: image

why?

File: src/Accordion/README.md [hunk 3]
```diff
@@ -6,51 +6,42 @@ Accordion component implemented (mostly) according to [WAI-ARIA specification](h
...
+    const newExpanded = [...expanded]
+    newExpanded[index] = !newExpanded[index]
+    setExpanded(newExpanded)
+  }
   return (
     <div style={{ height: 400 }}>
-      <Accordion sections={sections} onToggle={onToggle} />
+      <Accordion expanded={expanded} onToggle={onToggle}>
+        <AccordionSection title="Section 1">
+          {() => (
```
Comment (TejasQ) [inline src/Accordion/README.md]: Why is this a function? Why can't we just have a ReactNode in here?

File: src/AccordionSection/AccordionSection.tsx [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+  title,
+  children,
+  isMouseRef,
+}) => {
+  const titleId = `${id}-${index}-heading`
+  const contentId = `${id}-${index}-panel`
+  const [focusFlag, setFocusFlag] = React.useState(false)
+
+  if (expanded === undefined || index === undefined || isMouseRef === undefined || toggleSection === undefined) {
+    throw new Error("Use AccordionSection inside Accordion")
```
Comment (TejasQ) [inline src/AccordionSection/AccordionSection.tsx]:
```suggestion
    throw new Error("Only AccordionSections can be used inside Accordion. See https://operational-ui.netlify.com/#!/Accordion for more info.")
```

File: src/Accordion/README.md (hunk 2 above)
Comment (stereobooster) [reply src/Accordion/README.md]: Can you be more specific?

File: src/Accordion/README.md (hunk 3 above)
Comment (stereobooster) [reply src/Accordion/README.md]: we can have both

File: src/Accordion/README.md (hunk 2 above)
Comment (TejasQ) [reply src/Accordion/README.md]: I do not understand this commit: https://github.com/contiamo/operational-ui/pull/1039/commits/728f18ffce5c595dd073308faeb38b58bc879de1

It says TS breaks the example and removes the type annotation from `index`. But from my screenshot this works.

Why did you remove it with that commit?

File: src/Accordion/README.md (hunk 3 above)
Comment (TejasQ) [reply src/Accordion/README.md]: Can we keep the README example simple and remove it in that case?

File: src/Accordion/README.md (hunk 2 above)
Comment (stereobooster) [reply src/Accordion/README.md]: You edit directly in playground right? Try to edit readme itself and you will get screen like @fabien0102 posted above.

image

File: src/Accordion/README.md (hunk 3 above)
Comment (stereobooster) [reply src/Accordion/README.md]: sure thing, I thought I would show that you can use both (rows bellow use ReactNode)

File: src/Accordion/README.md (hunk 2 above)
Comment (TejasQ) [reply src/Accordion/README.md]: Interesting.

File: src/AccordionSection/AccordionSection.tsx [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+import styled from "../utils/styled"
+import { ChevronUpIcon, ChevronDownIcon } from "../Icon/Icon"
+import { DefaultProps } from "../types"
+import isFunction from "lodash/isFunction"
+
+export interface AccordionSectionProps extends DefaultProps {
+  title: React.ReactNode
+  children: (() => React.ReactNode) | React.ReactNode
+  // for internal use
+  index?: number
```
Comment (TejasQ) [inline src/AccordionSection/AccordionSection.tsx]: Maybe prefix with an `_`?

```suggestion
  _index?: number
```

**Pull request merged**